src/Common/src/Interop/Windows/BCrypt/Interop.Blobs.cs

```diff
@@ -14,23 +14,54 @@ internal partial class Interop
     internal partial class BCrypt
     {
         /// <summary>
+        ///     Append "value" to the data already in blob.
+        /// </summary>
+        internal static void Emit(byte[] blob, ref int offset, byte[] value)
+        {
+            Buffer.BlockCopy(value, 0, blob, offset, value.Length);
+            offset += value.Length;
+        }
+
+        /// <summary>
+        ///     Peel off the next "count" bytes in blob and return them in a byte array.
+        /// </summary>
+        internal static byte[] Consume(byte[] blob, ref int offset, int count)
+        {
+            byte[] value = new byte[count];
+            Buffer.BlockCopy(blob, offset, value, 0, count);
+            offset += count;
+            return value;
+        }
+
+        /// <summary>
         ///     Magic numbers identifying blob types
         /// </summary>
         internal enum KeyBlobMagicNumber : int
         {
             BCRYPT_ECDH_PUBLIC_P256_MAGIC = 0x314B4345,
+            BCRYPT_ECDH_PRIVATE_P256_MAGIC = 0x324B4345,
             BCRYPT_ECDH_PUBLIC_P384_MAGIC = 0x334B4345,
+            BCRYPT_ECDH_PRIVATE_P384_MAGIC = 0x344B4345,
             BCRYPT_ECDH_PUBLIC_P521_MAGIC = 0x354B4345,
+            BCRYPT_ECDH_PRIVATE_P521_MAGIC = 0x364B4345,
+            BCRYPT_ECDH_PUBLIC_GENERIC_MAGIC = 0x504B4345,
+            BCRYPT_ECDH_PRIVATE_GENERIC_MAGIC = 0x564B4345,
+
             BCRYPT_ECDSA_PUBLIC_P256_MAGIC = 0x31534345,
+            BCRYPT_ECDSA_PRIVATE_P256_MAGIC = 0x32534345,
             BCRYPT_ECDSA_PUBLIC_P384_MAGIC = 0x33534345,
+            BCRYPT_ECDSA_PRIVATE_P384_MAGIC = 0x34534345,
             BCRYPT_ECDSA_PUBLIC_P521_MAGIC = 0x35534345,
+            BCRYPT_ECDSA_PRIVATE_P521_MAGIC = 0x36534345,
+            BCRYPT_ECDSA_PUBLIC_GENERIC_MAGIC = 0x50444345,
+            BCRYPT_ECDSA_PRIVATE_GENERIC_MAGIC = 0x56444345,
+
```

Comment:
(Though I just realized this is the Windows implementation, not the Unix one, in which case it's less of an issue.  Nevermind.)